plugins/package-managers/cargo/src/main/kotlin/Cargo.kt

```diff
@@ -191,14 +192,19 @@ class Cargo(
         )
 
         val nonProjectPackages = packageById.values.mapNotNullTo(mutableSetOf()) { cargoPkg ->
-            cargoPkg.takeUnless { it.isProject() }?.toPackage(hashes)
+            cargoPkg.takeUnless { it.isProject(workingDir) }?.toPackage(hashes)
         }
 
         return listOf(ProjectAnalyzerResult(project, nonProjectPackages))
     }
 }
 
-private fun CargoMetadata.Package.isProject() = source == null
+private fun CargoMetadata.Package.getLocalPath(): File? =
+    id.substringAfter("path+file://", "").ifEmpty { null }
+        ?.removeSuffix(")")?.substringBefore("#")?.let { File(it) }
+
+private fun CargoMetadata.Package.isProject(root: File): Boolean =
```

Comment:
Looking at the signature, I have difficulties understanding what is expected to be passed as `root`. 
Can you help me understanding it? ..and can the name (root) be made more speaking ?

Reply:
I've added docs to clarify.